cmake/OpenCVPCHSupport.cmake

```diff
@@ -56,6 +56,10 @@ MACRO(_PCH_GET_COMPILE_FLAGS _out_compile_flags)
           endforeach()
         endif()
 
+        GET_TARGET_PROPERTY(_cxx_standard ${_PCH_current_target} CXX_STANDARD)
+        if (_cxx_standard)
+            LIST(APPEND ${_out_compile_flags} "-std=gnu++${_cxx_standard}")
```

Comment:
CMake has these variables, so hard-coding "gnu++" is not a good idea:
```
CMAKE_CXX11_EXTENSION_COMPILE_OPTION=-std=gnu++11
CMAKE_CXX11_STANDARD_COMPILE_OPTION=-std=c++11
CMAKE_CXX14_EXTENSION_COMPILE_OPTION=-std=gnu++14
CMAKE_CXX14_STANDARD_COMPILE_OPTION=-std=c++14
CMAKE_CXX98_EXTENSION_COMPILE_OPTION=-std=gnu++98
CMAKE_CXX98_STANDARD_COMPILE_OPTION=-std=c++98
```

Reply:
Thank you for the quick response I was not aware of `CMAKE_CXX11_EXTENSION_COMPILE_OPTION` it's definitely better than hard coding `-std=gnu++`.  I will shortly update the PR.